toolkit/components/uniffi-bindgen-gecko-js/src/templates/UniFFIScaffolding.cpp

```diff
@@ -29,17 +29,15 @@ extern "C" {
 {%- for ci in ci_list %}
 {%- for object in ci.object_definitions() %}
 {%- let pointer_type = ci.pointer_type(object) %}
-class {{ pointer_type }} : public UniFFIPointerType {
-    public:
-        static {{ pointer_type }}& getInstance() {
-            static {{ pointer_type }} instance;
-            return instance;
-        }
-    private:
-    {{ pointer_type }}() {
-            typeName = "{{ "{}::{}"|format(ci.namespace(), object.name()) }}"_ns;
-            destructor = {{ object.ffi_object_free().rust_name() }};
-        }
+class {{ pointer_type }} {
+  public:
+    static mozilla::uniffi::UniFFIPointerType* GetInstance() {
+      static mozilla::uniffi::UniFFIPointerType instance{
+        "{{ "{}::{}"|format(ci.namespace(), object.name()) }}"_ns,
+        {{ object.ffi_object_free().rust_name() }}
+      };
```

Comment:
Could this be simplified with less nesting?  It seems like you could have a top-level function instead of a class with 1 static method.  Or maybe just the `UniFFIPointerType` as a static variable.

Reply:
Simplified this a little!

I was thinking of having no class, and only a `UniFFIPointer` static variable, but I was fighting with the `ScaffoldingConverter` template - I decided to keep the class (and have it have an associated static variable) so we can take advantage of it in the template